### PR TITLE
docs: follow-up fixes from overhaul

### PR DESCRIPTION
**Pull request by mjlbach**
Please comment here with follow-up fixes, I will merge soon otherwise.

### CODE REVIEW

File: doc/lspconfig.txt [hunk 1]
```diff
@@ -11,8 +11,8 @@ TABLE OF CONTENTS                                                *lspconfig-toc*
       6a.  Adding Servers      (|lspconfig-adding-servers|)
       6b.  Index               (|lspconfig-index|)
 7.  Root directories           (|lspconfig-root-directories|)
-      6a. Root composition     (|lspconfig-root-composition|)
-      6b. Single file support  (|lspconfig-single-file-support|)
+      7a. Root composition     (|lspconfig-root-composition|)
```
Comment (clason) [inline doc/lspconfig.txt:14]: Didn't you rename that section?

It should be `lspconfig-root-detection` or `lspconfig-root-dir`, and 7a. should be `lspconfig-root-advanced` (for the tag as well).

Comment (mjlbach) [reply doc/lspconfig.txt]: I think its fixed now

**Comment by mjlbach** [comment]
@clason I also did some rebranding in the readme (nvim-lspconfig -> lspconfig)

File: README.md [hunk 2]
```diff
@@ -22,9 +24,9 @@ These features are not implemented in this repo, but in Neovim core. See `:help
 
 ## Install
 
-* Requires [Neovim v0.5.0](https://github.com/neovim/neovim/releases/tag/v0.5.0) or [Nightly](https://github.com/neovim/neovim/releases/tag/nightly). Update Neovim and nvim-lspconfig before reporting an issue.
+* Requires [Neovim v0.5.0](https://github.com/neovim/neovim/releases/tag/v0.5.0) or [Nightly](https://github.com/neovim/neovim/releases/tag/nightly). Update Neovim and 'lspconfig' before reporting an issue.
```
Comment (clason) [inline README.md:27]: Should be 0.5.1 now? Or "`0.5.0` or higher (including Nightly)"?

File: README.md [hunk 3]
```diff
@@ -65,7 +67,7 @@ These features are not implemented in this repo, but in Neovim core. See `:help
 
 ## Automatically launching language servers
 
-In order to automatically launch a language server, lspconfig searches up the directory tree from your current buffer to find a file matching the `root_dir` pattern defined in each server's configuration.  For [pyright](doc/server_configurations.md#pyright), this is any directory containing ".git", "setup.py",  "setup.cfg", "pyproject.toml", or "requirements.txt").
+In order to automatically launch a language server, 'lspconfig' searches up the directory tree from your current buffer to find a file matching the `root_dir` pattern defined in each server's configuration.  For [pyright](doc/server_configurations.md#pyright), this is any directory containing ".git", "setup.py",  "setup.cfg", "pyproject.toml", or "requirements.txt").
```
Comment (clason) [inline README.md:70]: "For, e.g., [pyright]"?

File: README.md (hunk 2 above)
Comment (clason) [reply README.md:27]: Backticks for `lspconfig` (here and below)?

File: README.md [hunk 4]
```diff
@@ -83,7 +85,7 @@ For a full list of servers, see [server_configurations.md](doc/server_configurat
 
 ## Keybindings and completion
 
-nvim-lspconfig does not map keybindings or enable completion by default. Manual, triggered completion can be provided by neovim's built-in omnifunc. For autocompletion, a general purpose [autocompletion plugin](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) is required. The following example configuration provides suggested keymaps for the most commonly used language server functions, and manually triggered completion with omnifunc (\<c-x\>\<c-o\>).
+'lspconfig' does not map keybindings or enable completion by default. Manual, triggered completion can be provided by neovim's built-in omnifunc. For autocompletion, a general purpose [autocompletion plugin](https://github.com/neovim/nvim-lspconfig/wiki/Autocompletion) is required. The following example configuration provides suggested keymaps for the most commonly used language server functions, and manually triggered completion with omnifunc (\<c-x\>\<c-o\>).
```
Comment (clason) [inline README.md:88]: "manually triggered" (cf. lspconfig.txt)

Comment (clason) [reply README.md:88]: And "Neovim" :P

**Review by clason — CHANGES_REQUESTED** [review]

**Pull request merged**